core/coreunix/add.go

```diff
@@ -480,6 +403,9 @@ func outputDagnode(out chan<- interface{}, name string, dn ipld.Node) error {
 		return err
 	}
 
+	name = strings.TrimPrefix(name, rootFileName)
+	name = strings.TrimLeft(name, "/")
```

Comment:
Moderately icky...

Reply:
Actually, this whole "rootFileName" thing is kind of broken (for now at least). Your point about not using MFS for a single file is probably the right way to go.

Otherwise, we'll need to correctly strip the root filename everywhere we send out an event.